### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by jaimeh**
**What question(s) did you decide to work on as a team?**

Which breweries are the most likely to be visited among Strava members who cycle?

**What is your data source(s)?**

Yelp (https://www.yelp.com/dataset_challenge), 
OpenStreetMaps (http://wiki.openstreetmap.org/wiki/API_v0.6#Retrieving_map_data_by_bounding_box:_GET_.2Fapi.2F0.6.2Fmap),
Strava (https://strava.github.io/api/)

These will meet the 3Vs. A variety of data, velocity comes from the GPS coordinates from each ride, and the size currently exceeds 3GB.

**How long does it take for you to download data? Have you download complete data set?**

The Yelp Raw Dataset took around 10minutes.
The Yelp Los Angeles Breweries CSV extracted and compressed from raw dataset took 3 minutes 

The OSM maps took about an hour to download from the API

The Strava data takes about 30 min to download a bounding box of data.

**How large is your data (size wise and number of records wise)?**
the map data is 1.6 gb 

hw2osmsize

The size of the Yelp Raw Dataset is 2,335,010 kb.
The size of the Yelp Los Angeles Breweries CSV is 12kb.
filesizeyelp

_
**Do you face any dirty data issue? If you do, how did you clean up your data?**

Strava library for the API auto munges the data. But the problem with the data is not how dirty it is but that it is very complicated to get the content that we need. We need to get the popular segments that are ridden within a bounding box and get the leaderboard. From the leaderboard we get the associated activites and from those activities we can finally get the GPX data for one specific ride. Each GPX data element is about .5MB. The data increases quickly. The collector is not completed because of the complicated nature of getting the specific data that we need. We need to reference and back reference different data objects.

The OSM XML file is complicated to process at the moment. We haven't found a good library to process the data that we have downloaded. We are looking at an algorithm that can find routes to use for collecting nearby street data.

**How do you store the data you downloaded?**

The Yelp data is stored in CSV format.
The OpenStreetMaps is stored in _To Be Answered_
Strava API data is stored in MongoDB

### CODE REVIEW

**Comment by yhoang001** [comment]
**What question(s) did you decide to work on as a team?**

Which breweries are the most likely to be visited among Strava members who cycle?

**What is your data source(s)?**

Yelp (https://www.yelp.com/dataset_challenge), 
OpenStreetMaps (http://wiki.openstreetmap.org/wiki/API_v0.6#Retrieving_map_data_by_bounding_box:_GET_.2Fapi.2F0.6.2Fmap),
Strava (https://strava.github.io/api/)

**How long does it take for you to download data? Have you download complete data set?**





    Please provide a screenshot on your total data size

**How large is your data (size wise and number of records wise)?**
the map data is 1.6 gb 

hw2osmsize

The size of the data is: _To be answered
_
**Do you face any dirty data issue? If you do, how did you clean up your data?**

_To be answered_

**How do you store the data you downloaded?**

The Yelp data is stored in CSV format.
The OpenStreetMaps is stored in _To Be Answered_
Strava API data is stored in MongoDB

File: src/main/java/edu/csula/datascience/acquisition/StravaAthleteCollector.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    public void save(Collection<StravaAthlete> athlete) {
+        List<Document> documents = athlete.stream()
+                .map(item -> new Document()
+                        .append("id", item.getId())
+                        .append("firstname", item.getFirstname())
+                        .append("lastname", item.getLastname())
+                        .append("city", item.getCity())
+                        .append("state", item.getState())
+                        .append("country", item.getCountry()))
+                .collect(Collectors.toList());
```
Comment (rcliao) [inline src/main/java/edu/csula/datascience/acquisition/StravaAthleteCollector.java:54]: Nice use of Java 8 streaming api.

File: src/main/java/edu/csula/datascience/acquisition/StravaAthleteSource.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return false;
+    }
+
+    @Override
+    public Collection<StravaAthlete> next() {
+        List<StravaAthlete> list = Lists.newArrayList();
+
+        AuthorisationService service = new AuthorisationServiceImpl();
+        // Authorize through
+        Token token = service.tokenExchange(11245, "752834d5dd91633194a8a97ad4af594594ea11e0", "33da4461dcb4e76c312b24f47c731a3956f1d469");
```
Comment (rcliao) [inline src/main/java/edu/csula/datascience/acquisition/StravaAthleteSource.java:36]: Token ... _ahem_ should not be stored in source code.

**Comment by rcliao** [comment]
In terms of the acquisition and storage, they looks fine.

Since you guys are using geo location variable. It might be interesting for you guys to look at elastic search geo location data. Take a look at that and see if you guys can do your initial EDA with Kibana.

Overall code looks clean.

10/10 + bonus 2

**Pull request merged**